appservice/src/main/java/nu/yona/server/messaging/rest/MessageController.java

```diff
@@ -141,40 +140,32 @@ public MessageActionResource(MessageActionDTO messageAction, UUID userID)
 		}
 	}
 
-	static class MessageResource extends Resource<MessageDTO>
-	{
-		public MessageResource(MessageDTO message)
-		{
-			super(message);
-		}
-	}
-
-	private static class MessageResourceAssembler extends ResourceAssemblerSupport<MessageDTO, MessageResource>
+	private static class MessageResourceAssembler extends ResourceAssemblerSupport<MessageDTO, MessageDTO>
 	{
 		private UUID userID;
 
 		public MessageResourceAssembler(UUID userID)
 		{
-			super(MessageController.class, MessageResource.class);
+			super(MessageController.class, MessageDTO.class);
 			this.userID = userID;
 		}
 
 		@Override
-		public MessageResource toResource(MessageDTO message)
+		public MessageDTO toResource(MessageDTO message)
 		{
-			MessageResource messageResource = instantiateResource(message);
+			message.removeLinks();
```

Comment:
In the past, we were wrapping the DTOs in resources and adding the links there. Now we add the links to the DTOs. In my prototype, I reused the DTOs and then added the links multiple times. To prevent that, I clear the links firs, so we are sure it only has what we add here.